src/ClockMock.php

```diff
@@ -223,7 +223,13 @@ public static function mock_date_create_immutable_from_format(): callable
             // Create an immutable instance starting from the mutable mock, so we don't have to replicate mocking logic.
             $mutableDateTime = date_create_from_format($format, $datetime, $timezone);
 
-            return new \DateTimeImmutable($mutableDateTime->format('Y-m-d\TH:i:s.u'), $mutableDateTime->getTimezone());
+            return $mutableDateTime
```

Comment:
instead of treating a valid date object like a truthy value in a ternary statement, I believe it would be more explicit to make an early return if the return value from `date_create_from_format` is false, like that:

```
if ($mutableDateTime === false) {
    return false;
}
```

Also, a line with a comment saying that this variable can be `false` in case of invalid format would help.